packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -73,17 +88,29 @@ class TransferEngine implements ITransferEngine {
     ]);
   }
 
-  async integrityCheck(): Promise<boolean> {
+  async #resolveProviderResource() {
```

Comment:
Since this is just metadata, should we call it Metadata instead of Resource?

Reply:
I'll update this when the Christian's Schema validation PR will be merged & I'll add the schema loading there too